### DISCUSSION TITLE
How to tell the current thread?

### DISCUSSION THREAD

**Discussion by nnnnt21** [Minestom/Minestom#2216] [Help]
What thread runs the events?

If EntityAttackEvent.class is called from EntityCreature.attack what thread runs it? The attacker's thread? The thread that called EntityCreature.attack? The target entity? 

How can I check this, how can I figure out what thread is running it?

**Comment by Tofaa2** [comment]
Thread#currentThread()

Events are ran in the thread that theyre called in
